consensus/src/block_storage/sync_manager.rs

```diff
@@ -304,9 +320,8 @@ impl BlockRetriever {
 
     /// Retrieve n blocks for given block_id from peers
     ///
-    /// Returns Result with Vec that has a guaranteed size of num_blocks
-    /// This guarantee is based on BlockRetrievalResponse::verify that ensures that number of
-    /// blocks in response is equal to number of blocks requested.  This method will
+    /// Returns Result with Vec that if succeeded, contains the target block if provided,
+    /// or at least contains `num_blocks` many blocks. This method will
```

Comment:
this function should always return Ok([target block..block]) or Err, chunking should be handled internally